Issue #2371861 by DuaelFr, YesCT, pietmarcus, Gábor Hojtsy, tucho: Strings including tokens in href or src attributes cannot be translated due to safeness check incompatibilities

jrearick opened the pull request:
https://github.com/backdrop/backdrop-issues/issues/2013

This PR includes both committed patches from Issue #2371861

klonos commented on core/modules/locale/tests/locale.test:
```diff
@@ -2808,3 +2808,46 @@ class LocaleLanguageNegotiationInfoFunctionalTest extends BackdropWebTestCase {
+/**
+ * Tests locale translation safe string handling.
+ */
+class LocaleStringIsSafeTest extends DrupalWebTestCase {
```
Not sure, but I think that this needs to be `BackdropWebTestCase` instead (?). The only places we use `DrupalWebTestCase` seem to be in https://github.com/backdrop/backdrop/blob/1.x/core/includes/bootstrap.classes.inc#L130 and https://github.com/backdrop/backdrop/blob/1.x/core/includes/drupal.classes.inc#L41 and I believe that these are in place to provide wrapper/BC. Not an expert coder though, so take my words with a grain of salt.

jrearick replied:
Good catch. I am also not an expert coder, but what you say make sense to me. I'll see if I can change it to extend `BackdropWebTestCase` and see what happens with and without compatibility enabled.

quicksketch commented on core/modules/locale/tests/locale.test:
```diff
@@ -2808,3 +2808,46 @@ class LocaleLanguageNegotiationInfoFunctionalTest extends BackdropWebTestCase {
+      'description' => 'Tests locale translation safe string handling.',
+      'group' => 'Locale',
+    );
+  }
```
This test isn't going to be called because Backdrop uses *.tests.info files to identify test classes rather than `getInfo()` methods. Could you put this in `locale.tests.info` and remove this method?

jrearick commented:
Thanks for the good feedback @quicksketch. I added the test to the tests.info file and was able to run it from the testing UI. I updated my branch to reflect that change.

quicksketch commented:
Killer. Thanks! Merged into 1.x and 1.4.x for 1.4.4.
